charts/partials/templates/_proxy-init.tpl

```diff
@@ -46,7 +46,7 @@ securityContext:
     {{- include "partials.proxy-init.capabilities.drop" . | nindent 4 -}}
     {{- end }}
     {{- end }}
-  {{- if .Values.proxyInit.closeWaitTimeoutSecs }}
+  {{- if or .Values.proxyInit.closeWaitTimeoutSecs .Values.proxyInit.runAsRoot }}
```

Comment:
I think we need to also change [L32-L34](https://github.com/linkerd/linkerd2/blob/94c04fd06ac62d09b563c3d515580c5328dd9841/charts/partials/templates/_proxy-init.tpl#L32-#L34) here.

When I try to install with `runAsRoot=true`, I get errors for all control plane deployments; `allowPrivilegeEscalation` cannot be set to false when `privileged` is set to true. I think if we add the same conditional to L32 we should be good to go.

Error message, to get this, I built your branch and tried to install linkerd through the CLI (`linkerd install --set "proxyInit.runAsRoot=true" | kubectl apply -f -`)
```
Error from server (Invalid): error when creating "STDIN": Deployment.apps "linkerd-proxy-injector" is invalid: spec.template.spec.initContainers[0].securityContext: Invalid value: core.SecurityContext{Capabilities:(*core.Capabilities)(0xc01a587260), Privileged:(*bool)(0xc02460f5aa), SELinuxOptions:(*core.SELinuxOptions)(nil), WindowsOptions:(*core.WindowsSecurityContextOptions)(nil), RunAsUser:(*int64)(0xc02460f5d0), RunAsGroup:(*int64)(nil), RunAsNonRoot:(*bool)(0xc02460f5ac), ReadOnlyRootFilesystem:(*bool)(0xc02460f5ab), AllowPrivilegeEscalation:(*bool)(0xc02460f57f), ProcMount:(*core.ProcMountType)(nil), SeccompProfile:(*core.SeccompProfile)(nil)}: cannot set `allowPrivilegeEscalation` to false and `privileged` to true
```

Reply:
OK, I fixed that and also collapsed the two `if` statements into one to simplify things.